src/hnsw/build.c

```diff
@@ -91,6 +93,15 @@ static void AddTupleToUsearchIndex(ItemPointer tid, Datum *values, HnswBuildStat
     if(buildstate->usearch_index != NULL) {
         size_t capacity = usearch_capacity(buildstate->usearch_index, &error);
         if(capacity == usearch_size(buildstate->usearch_index, &error)) {
+            double             M = ldb_HnswGetM(index);
+            double             mL = 1 / log(M);
+            usearch_metadata_t meta = usearch_metadata(buildstate->usearch_index, &error);
+            uint32             node_size = UsearchNodeBytes(&meta, meta.dimensions * sizeof(float), (int)(mL + .5));
+            if(2 * usearch_size(buildstate->usearch_index, &error) * node_size
+               >= (size_t)maintenance_work_mem * 1024L) {
+                usearch_free(buildstate->usearch_index, &error);
+                elog(ERROR, "index size exceeded maintenance_work_mem during index construction");
```

Comment:
For now, let's leave this as a warning. 
We are carrying out fault tolerance tests now and we should throw errors like this once we are sure we handle them well.